lib/streamlit/elements/lib/file_uploader_utils.py

```diff
@@ -59,8 +59,9 @@ def enforce_filename_restriction(filename: str, allowed_types: Sequence[str]) ->
     enforce file type check by extension on the frontend, but we check it on backend
     before returning file to the user to protect ourselves.
     """
-    extension = os.path.splitext(filename)[1].lower()
-    if allowed_types and extension not in allowed_types:
+    base_name, extension = os.path.splitext(filename.lower())
+
+    if not any(filename.endswith(allowed_type) for allowed_type in allowed_types):
```

Comment:
@kajarenc Do we validate the allowed types starts with a dot? I am wondering if we should handle this case so that if `allowed_type` were like `tar.gz` it wouldn't allow `myfiletar.gz`

Reply:
@kmcgrady yes! the `normalize_upload_file_type` helper function always normalize allowed types to start from the `.` even if user not specified it